### DISCUSSION TITLE
Gitlab to Github migration

### DISCUSSION THREAD

**Discussion by kulsinghdeep** [community/community#52142] [Enterprise]
### Select Topic Area

Question

### Body

Hi team, can someone provide doc how can we move groups and user from gitlab to github.

**Comment by AneeqMalik** [comment]
Yes, I can help you with that. Here's a general overview of the steps you can follow to move groups and users from GitLab to GitHub:

Create an organization in GitHub: If you don't already have an organization in GitHub, create one.

Export GitLab data: In GitLab, go to the Admin Area > Settings > Export, and select "Export all groups, projects, and their data". This will create a .tar.gz file containing all the data that you want to transfer to GitHub.

Extract the exported data: Extract the .tar.gz file you downloaded from GitLab to a local directory.

Convert GitLab usernames to GitHub usernames: GitHub usernames are not case sensitive, whereas GitLab usernames are. You will need to convert all GitLab usernames to lowercase to ensure that they match GitHub usernames.

Import groups to GitHub: In GitHub, go to your organization's settings and click on the "Import Data" button. Select "Import a repository" and then select "Import a repository from a URL". Enter the URL of the GitLab group that you want to import and click on "Begin import". This will create a new repository in GitHub with the same name as the GitLab group.

Transfer repositories: After you have imported the groups, you can transfer the repositories from GitLab to GitHub. To do this, go to the repository in GitLab, click on the "Settings" button, and then select "Transfer". Enter the name of the GitHub organization you created in step 1 and select the repository you want to transfer. Click on "Transfer".

Add users to GitHub organization: In GitHub, go to your organization's settings and click on "People". Click on "Invite member" and enter the email address of the GitLab user you want to invite. Once the user accepts the invitation, they will be added to your GitHub organization.

These are the general steps to move groups and users from GitLab to GitHub. Note that there may be some specific differences depending on the version of GitLab and GitHub you are using, and the permissions and access levels of the users and groups involved. It's always a good idea to carefully review the documentation for both platforms before proceeding with the migration.

**Reply by kulsinghdeep** [reply]
Hi Thanks for your answer, am not seeing this option in Gitlab and Github. Are these available in free tier of Gitlab and Github or we have do this on Paid account.

**Reply by AneeqMalik** [reply]
You can try github student pack.

**Reply by nearlythere** [reply]
> Hi Thanks for your answer, am not seeing this option in Gitlab and Github. Are these available in free tier of Gitlab and Github or we have do this on Paid account.

Organisations are available in the free tier on GitHub, but limited. https://docs.github.com/en/organizations/collaborating-with-groups-in-organizations/about-organizations

**Reply by syedobaid123** [reply]
Thanks bro. Any idea about pipeline migration from gitlab to github? Since the pipeline format in gitlab ci is different and the pipeline format in github actions is different

**Comment by nearlythere** [comment]
I am looking into this myself now for a small migration. Curious what you learn! There is no 1:1 translation of GitLab groups in GitHub. 

Here’s some notes about GitLab groups v GitHub teams you might find helpful. 
 
In GitLab, groups are used to chunk together repositories, there's no similar feature in GitHub. And the permissions work a little differently compared to GitHub. In some respects GitLab Groups are like GitHub Organisations, but there's no notion of "sub-organisations" like Sub-teams in GitLab. So in other ways it's like Teams (in terms of granular access control.) 

Some notes: 
- In GitLab, you add projects to a group, whereas in GitHub, you add Teams to a repository to control access. 
- The groups in GitLab provide a namespace (like Organisations in GitHub). GitLab Docs https://docs.gitlab.com/ee/user/group/
- In GitHub, “Child teams inherit the parent's access permissions” https://docs.github.com/en/enterprise-cloud@latest/organizations/organizing-members-into-teams/about-teams#nested-teams 
 
This comparison of GitLab v GitHub in 2016, though I don’t know if it’s up to date anymore! 😃 
https://about.gitlab.com/blog/2016/01/27/comparing-terms-gitlab-github-bitbucket/

And this competitor comparison with GitHub doesn't really delve into the differences too much, since there isn't an analogous structure in GitHub. 
https://about.gitlab.com/competition/github/?feature=Source%20Code%20Management&stage=Create 

If you find out anything else, I'd be happy to hear what you figure out.

**Comment by tin-cat** [comment]
Here's a tool I created to specifically migrate issues from a GitLab repository to GitHub, it might come in handy: https://github.com/tin-cat/migrate-issues-gitlab-to-github
